accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,242 @@
+  - the SCC CDN
+  - a plain RMT Server ("from-mirror setup", still requires connection to SCC)
+  - a plain directory exported from an RMT Server ("disconnected setup")
+  - a custom repository
+
+Presently, it is not possible to sync content from a SUSE-operated Cloud RMT Server, because:
+  - a Uyuni Server does not come with "tokens" (unlike pay-as-you-go instances, see above)
+  - Uyuni does not know how to pass them to RMT Servers
+  - Uyuni Server do not know which repositories are served via such RMT Servers
+  - Uyuni does not know for which repos/products the user is entitled for - which pay-as-you-go instances of which products he run
+
+Since Uyuni cannot contact the cloud RMT servers the only option for users is to contact SUSE and get one SCC account which allows to synchronize content from the SCC CDN.
+The process for this is not straightforward and adds excessive complexity to users and SUSE teams.
+
+The goal for this RFC is to propose a solution to simplify this process and allow Uyuni to synchronize content directly from RMT cloud servers in a user friendly way.
+
+
+# Detailed design
+[design]: #detailed-design
+
+Our solution will be based on the following steps:
+  - Manage pay-as-you-go ssh connections data on Uyuni
+    - Register pay-as-you-go ssh connections data
+    - Update existing ssh connections data
+    - Delete pay-as-you-go ssh connections data
+  - Develop a new taskomatic task and job schedule to:
+    - Retrieve repositories and authentication data from the pay-as-you-go instance
+    - Register repositories and authentication data on Uyuni
+  - Teach Uyuni reposync how cloud RMT authentication data to synchronize product repositories
+
+
+With this solution the expected user flow would be:
+  - Provide to Uyuni ssh information to connect to the pay-as-you-go instance
+    - uyuni will save this information on the database
+    - start a single job execution to synchronize repositories and cloud RMT authentication data
+    - product loaded from the pay-as-you-go instance will be available to import after the task finishes
+  - Import product using existing "add products" feature (available at UI, API, and cmd)
+    - reposync will be able to download all needed data from cloud RMT server
+  - Bootstrap instances using the existing methods
+
+## Manage pay-as-you-go ssh connection data
+
+User should be able to register and manage ssh connection data to pay-as-you-go instances.
+Connecting to those instances, uyuni should extract repository information and authentication data.
+Users should be able to manage ssh connection data using the web UI or the XML-RPC API.
+
+Supported operations:
+  - Add new ssh connection data for pay-as-you-go
+    - should trigger an execution of the taskomatic task to extract data from the payg-as-you-go instance
+  - Remove ssh connection data
+    - Associated data in `suseCredentials` and `susesccrepositoryauth` should also be removed
+  - Update ssh connection data
```

Comment:
Should the task be triggered automatically also in this case?

Reply:
yes, it should trigger it